### ISSUE TITLE
[css-animations][css-transitions][web-animations] Introduce a host-syntax for color interpolation

### ISSUE THREAD

**Issue by lilles** [w3c/csswg-drafts#7741]
css-color-4 introduces the concept of a [host-syntax](https://www.w3.org/TR/css-color-4/#host-syntax) for color interpolation. For legacy colors, interpolation happens in sRGB for back compat reasons. When the relevant specs do not specify their own host-syntax for interpolation spaces, it falls back to OKLab for non-legacy colors.

We probably want to introduce a host-syntax for transitions and animations for explicitly specifying an interpolation space, and possibly also specify if the default interpolation space depends on which color space the endpoints or keyframe values are specified in.

See also issue #7567.

@mysteryDate

**Comment by svgeesus** [comment]
Right, we need to add a [`<color-interpolation-method> `](https://www.w3.org/TR/css-color-4/#color-interpolation-method) to the grammar for those.

> and possibly also specify if the default interpolation space depends on which color space the endpoints or keyframe values are specified in.

I think that is [already covered](https://www.w3.org/TR/css-color-4/#color-interpolation-method):

> If the host syntax does not define what color space interpolation should take place in, it defaults to OKLab.

> However, user agents may handle interpolation between legacy sRGB color formats (hex colors, named colors, [rgb()](https://www.w3.org/TR/css-color-4/#funcdef-rgb), [hsl()](https://www.w3.org/TR/css-color-4/#funcdef-hsl) or [hwb()](https://www.w3.org/TR/css-color-4/#funcdef-hwb) and the equivalent alpha-including forms) in gamma-encoded sRGB space. This provides Web compatibility; legacy sRGB content interpolates in the sRGB space by default.

> This also means that authors can choose to opt-in to better interpolation, even between sRGB colors, by using the non-legacy color(srgb r g b) form for at least one of their colors.

**Comment by LeaVerou** [comment]
How about `transition-color-space: auto | <color-interpolation-method>` with initial value `auto` which is whatever the UA does by default? Same for animations.

**Comment by lilles** [comment]
> How about `transition-color-interpolation: auto | <color-interpolation-method>` with initial value `auto` which is whatever the UA does by default? Same for animations.

Should it take multiple values like other transition/animation properties? That is, should it be possible to transition color and background-color with two different methods on the same element?

**Comment by LeaVerou** [comment]
> > How about `transition-color-interpolation: auto | <color-interpolation-method>` with initial value `auto` which is whatever the UA does by default? Same for animations.
> 
> Should it take multiple values like other transition/animation properties? That is, should it be possible to transition color and background-color with two different methods on the same element?

Correct. I updated my earlier comment.

**Comment by birtles** [comment]
If we're supporting multiple values, that sounds like this is a per-animation property (as opposed to per-element like it is in SVG).

In that case, I guess this should also be exposed on the underlying `Animation` JS object too.

Would this same property also affect how colors are [added](https://drafts.csswg.org/css-values-4/#addition) together? Or are colors [not additive](https://drafts.csswg.org/css-values-4/#not-additive)?

**Comment by svgeesus** [comment]
> Would this same property also affect how colors are [added](https://drafts.csswg.org/css-values-4/#addition) together? Or are colors [not additive](https://drafts.csswg.org/css-values-4/#not-additive)?

We [resolved](https://github.com/w3c/csswg-drafts/issues/8576#issuecomment-1570575637) that colors are not additive. [Values 4 is up to date there](https://drafts.csswg.org/css-values-4/#combine-colors)

**Comment by svgeesus** [comment]
Realizing  this issue is a dup of

 - https://github.com/w3c/csswg-drafts/issues/7063

where there is already [a resolution to add control of colorspace used for transitioning colors](https://github.com/w3c/csswg-drafts/issues/7063#issuecomment-1561605996) from May 2023
